staging/src/k8s.io/api/apps/v1/types.go

```diff
@@ -520,12 +536,18 @@ type DeploymentConditionType string
 // These are valid conditions of a deployment.
 const (
 	// Available means the deployment is available, ie. at least the minimum available
-	// replicas required are up and running for at least minReadySeconds.
+	// replicas required are up and running for at least minReadySeconds. A deployment
+	// that allows a minimum availability of zero will be considered available even if
+	// no available replicas exist. A deployment will have a minimum availability of
+	// zero when using the Recreate strategy, or when setting maxUnavailable in
+	// RollingUpdate deployments greater than or equal to the number of replicas, or
+	// when a RollingUpdate deployment has only one replica and any non-zero
+	// maxUnavailable value.
```

Comment:
Is it useful to have 2 definitions? Don't we want to make only one definition of Available and then copy it to the Conditions doc?

Reply:
Since few people reference the constant out of tree, i think we could have the authoritative definition on Conditions and have a “see type DeploymentSpecConditions” for all the constants.